### DISCUSSION TITLE
Pitfall of ProgressRing visibility

### DISCUSSION THREAD

**Discussion by ArchieCoder** [microsoft/microsoft-ui-xaml#9500] [General]
Earlier this week, I thought there was a bug in the WrapPanel control. My WrapPanel contains buttons and a ProgressRing.

My common usage when I add a ProgressRing is using a binding on the IsActive property. As you would expect, when IsActive is False, the control is now shown.

When resizing my view that contains the WrapPanel, I had the bug that there was a point where the WrapPanel wrapped into 2 lines where the 2nd line was empty! This was obviously not great.

When I dig more, I found out that I was missing the Visibility on the control.

When IsActive is False (not shown), the control size retains its full size.

Would it be better to have a zero size when IsActive is False?

**Comment by kmgallahan** [comment]
> Would it be better to have a zero size when IsActive is False?

No. `Visibility=Collapsed` means collapsed. If the control collapses as a side effect of `IsActive=false` then others will show up to complain that the ProgressRing should retain its size when `IsActive=false` for their scenarios.

**Reply by marcelwgn** [reply]
I can only add what @kmgallahan said, the behavior is correct and it's also called out in the [remarks section of the control](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.progressring?view=windows-app-sdk-1.5#remarks):

> [...] Set the [IsActive](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.progressring.isactive?view=windows-app-sdk-1.5#microsoft-ui-xaml-controls-progressring-isactive) property to turn the ProgressRing on or off. If [IsActive](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.progressring.isactive?view=windows-app-sdk-1.5#microsoft-ui-xaml-controls-progressring-isactive) is false, the ProgressRing is not shown, but space is reserved for it in the UI layout. To not reserve space for the ProgressRing, set its [Visibility](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.visibility) property to [Collapsed](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.visibility).
